src/Illuminate/View/View.php

```diff
@@ -58,7 +58,7 @@ class View implements ArrayAccess, ViewContract
      * @param  \Illuminate\View\Engines\EngineInterface  $engine
      * @param  string  $view
      * @param  string  $path
-     * @param  array   $data
+     * @param  mixed   $data
```

Comment:
`\Illuminate\Contracts\Support\Arrayable|object|array` => `mixed`
As for https://github.com/laravel/framework/blob/5.2/src/Illuminate/Support/Collection.php#L1135-L1156